docs/source/sparse.rst

```diff
@@ -1,146 +1,339 @@
-     0  0  3
-     4  0  5
-    [torch.FloatTensor of size 2x3]
+.. _sparse-hybrid-docs:
 
-Note that the input to LongTensor is NOT a list of index tuples.  If you want
+Hybrid tensors
+++++++++++++++
+
+Torch supports hybrid sparse tensors with dimensions split into sparse
+dimensions and dense dimensions. Dense dimensions always follow
+sparse dimensions, that is, mixing of dense and sparse dimensions is
+not supported. One can interpret hybrid tensors as sparse tensors with
+elements being strided tensors. We use (M + K)-dimensional tensor to
+denote a N-dimensional hybrid sparse tensor where M and K are the
+number of sparse and dense dimensions, respectively, such that M + K
+== N holds.
+
+See :ref:`Hybrid COO tensors <sparse-hybrid-coo-docs>`.
+
+Sparse tensor formats
++++++++++++++++++++++
+
+Torch supported sparse storage formats are:
+
+- COO(rdinate) format where tensor elements are stored using the lists
+  of indices and the corresponding values. Unspecified elements have
+  values equal to the fill value. To construct a sparse tensor in COO
+  format, use :func:`torch.sparse_coo_tensor` function.
+
+.. _sparse-coo-docs:
+
+Sparse COO tensors
+******************
+
+The sparse COO tensor is represented as a pair of dense tensors: a 2D
+tensor of indices and a tensor of values. A sparse COO tensor can be
+constructed by providing these two tensors, as well as the size of the
+sparse tensor (which cannot be inferred from the indices and values
+tensors!). Suppose we want to define a sparse tensor with the entry 3
+at location (0, 2), entry 4 at location (1, 0), and entry 5 at
+location (1, 2).  We would then write:
+
+    >>> i = [[0, 1, 1],
+             [2, 0, 2]]
+    >>> v =  [3, 4, 5]
+    >>> s = torch.sparse_coo_tensor(i, v, (2, 3))
+    >>> s
+    tensor(indices=tensor([[0, 1, 1],
+                           [2, 0, 2]]),
+           values=tensor([3, 4, 5]),
+           size=(2, 3), nnz=3, layout=torch.sparse_coo)
+    >>> s.to_dense()
+    tensor([[0, 0, 3],
+            [4, 0, 5]])
+
+Note that the input ``i`` is NOT a list of index tuples.  If you want
 to write your indices this way, you should transpose before passing them to
 the sparse constructor:
 
-    >>> i = torch.LongTensor([[0, 2], [1, 0], [1, 2]])
-    >>> v = torch.FloatTensor([3,      4,      5    ])
-    >>> torch.sparse.FloatTensor(i.t(), v, torch.Size([2,3])).to_dense()
-     0  0  3
-     4  0  5
-    [torch.FloatTensor of size 2x3]
-
-You can also construct hybrid sparse tensors, where only the first n
-dimensions are sparse, and the rest of the dimensions are dense.
-
-    >>> i = torch.LongTensor([[2, 4]])
-    >>> v = torch.FloatTensor([[1, 3], [5, 7]])
-    >>> torch.sparse.FloatTensor(i, v).to_dense()
-     0  0
-     0  0
-     1  3
-     0  0
-     5  7
-    [torch.FloatTensor of size 5x2]
+    >>> i = [[0, 2], [1, 0], [1, 2]]
+    >>> v =  [3,      4,      5    ]
+    >>> s = torch.sparse_coo_tensor(list(zip(*i)), v, (2, 3))         # or
```

Comment:
nit: how does the "or" renders in the final doc here? Does it look good?

Reply:
![Screenshot from 2020-09-30 19-41-44](https://user-images.githubusercontent.com/402156/94714731-374f9d00-0355-11eb-9a6c-169d265e5e1d.png)

Is that ok?

Reply:
Maybe I would put it in a new line between the two then that says:
```
>>> # Or another equivalent formulation to get s
```
To be extra clear that these two lines are equivalent?

Reply:
Done.